lib/components/PoolUI.jsx

```diff
@@ -41,44 +62,49 @@ export const PoolUI = (props) => {
   const provider = walletContext.state.provider
   const usersAddress = walletContext._onboard.getState().address
 
-  const [ethBalance, setEthBalance] = useState(ethers.utils.bigNumberify(0))
-  const [poolAddresses, setPoolAddresses] = useState({
-    prizePool,
-  })
-  const [genericChainValues, setGenericChainValues] = useState({
-    loading: true,
-    tokenSymbol: 'TOKEN',
-    poolTotalSupply: '1234',
-  })
-
-  const [usersChainValues, setUsersChainValues] = useState({
-    loading: true,
-    usersTicketBalance: ethers.utils.bigNumberify(0),
-    usersTokenAllowance: ethers.utils.bigNumberify(0),
-    usersTokenBalance: ethers.utils.bigNumberify(0),
-  })
+  const [ethBalance, setEthBalance] = useAtom(ethBalanceAtom)
+  const [prizePoolType] = useAtom(prizePoolTypeAtom)
+  const [poolAddresses, setPoolAddresses] = useAtom(poolAddressesAtom)
+  const [_usersAddress, setUsersAddress] = useAtom(usersAddressAtom)
+  const [erc20Awards, setErc20Awards] = useAtom(erc20AwardsAtom)
+  const [network, setNetwork] = useAtom(networkAtom)
+  const [poolChainValues, setPoolChainValues] = useAtom(poolChainValuesAtom)
+  const [userChainValues, setUserChainValues] = useAtom(userChainValuesAtom)
 
-  useInterval(() => {
-    fetchChainData(
-      networkName,
-      usersAddress,
-      poolAddresses,
-      setPoolAddresses,
-      setGenericChainValues,
-      setUsersChainValues
-    )
-  }, 25000)
+  useEffect(() => {
+    // TODO: Probably need to reset other atoms if this changes.
+    setPoolAddresses({
+      prizePool
+    })
+  }, [prizePool])
+
+  useEffect(() => {
+    setUsersAddress(usersAddress)
+  }, [usersAddress])
 
   useEffect(() => {
-    fetchChainData(
-      networkName,
-      usersAddress,
+    setNetwork({
+      name: networkName,
+      id: nameToChainId(networkName)
+    })
+  }, [networkName])
+
+  usePrizePoolType()
+  usePoolAddresses()
+  usePoolChainValues()
+  useExternalErc20Awards()
+
+  // Keep chain values fresh
+  useInterval(() => {
+    fetchPoolChainValues(provider, poolAddresses, prizePoolType, setPoolChainValues)
+    fetchUserChainData(provider, poolAddresses, usersAddress, setUserChainValues)
+    fetchErc20AwardBalances(
+      provider,
       poolAddresses,
-      setPoolAddresses,
-      setGenericChainValues,
-      setUsersChainValues
+      poolChainValues.externalErc20Awards,
+      setErc20Awards
     )
-  }, [provider, usersAddress, poolAddresses])
+  }, 25000)
```

Comment:
We should move these magic numbers to the constants file. In the `pool-app` I have constants called `MAINNET_POLLING_INTERVAL`, `UNISWAP_SUBGRAPH_POLLING_INTERVAL`, etc.